core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSourceTest.java

```diff
@@ -0,0 +1,425 @@
+  private static final ArtifactId BATCH_APP_ARTIFACT_ID =
+    NamespaceId.DEFAULT.artifact("etlbatch", CURRENT_VERSION.getVersion());
+  private static final ArtifactSummary ETLBATCH_ARTIFACT =
+    new ArtifactSummary(BATCH_APP_ARTIFACT_ID.getArtifact(), BATCH_APP_ARTIFACT_ID.getVersion());
+  private static final String SOURCE_FOLDER_PATH = "src/test/resources/xmlsource/";
+  private File sourceFolder = new File(SOURCE_FOLDER_PATH);
+  private File targetFolder = new File("src/test/resources/xmltarget/");
+  private String sourceFolderUri = sourceFolder.toURI().toString();
+  private String targetFolderUri = targetFolder.toURI().toString();
+
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", false);
+
+  @BeforeClass
+  public static void setupTest() throws Exception {
+    setupBatchArtifacts(BATCH_APP_ARTIFACT_ID, ETLBatchApplication.class);
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("core-plugins", "1.0.1"), BATCH_APP_ARTIFACT_ID,
+                      XMLReaderBatchSource.class);
+  }
+
+  /**
+   * Method to copy test xml files into source folder path, from where XMLReader read the file.
+   */
+  @Before
+  public void copyFiles() throws IOException {
+      String xmlSourceFolder = "src/test/resources/";
+      String catalogLargeFile = "catalogLarge.xml";
+      String catalogSmallFile = "catalogSmall.xml";
+      FileUtils.copyFile(new File(xmlSourceFolder + catalogLargeFile), new File(SOURCE_FOLDER_PATH + catalogLargeFile));
+      FileUtils.copyFile(new File(xmlSourceFolder + catalogSmallFile), new File(SOURCE_FOLDER_PATH + catalogSmallFile));
+  }
+
+  /**
+   * Method to clear source and target folders. This ensures that source and target folder are ready to use for next
+   * JUnit Test case.
+   */
+  @After
+  public void clearSourceAndTargetFolder() {
+    File sourceDirectory = sourceFolder;
+    if (sourceDirectory != null) {
+      File[] sourceFiles = sourceDirectory.listFiles();
+      if (sourceFiles != null && sourceFiles.length > 0) {
+        for (File sourceFile : sourceFiles) {
+          sourceFile.delete();
+        }
+      }
+    }
+
+    File targetDirectory = targetFolder;
+    if (targetDirectory != null) {
+      File[] targetFiles = targetDirectory.listFiles();
+      if (targetFiles != null && targetFiles.length > 0) {
+        for (File targetFile : targetFiles) {
+          targetFile.delete();
+        }
+      }
+    }
+  }
+
+  /**
+   * Method to Pre-Populate File tracking KeyValue with previously processed XML file.
+   */
+  private void createPreProcessedRecord(String processedFileTable, Date preProcessedDate) throws Exception {
+    DataSetManager<KeyValueTable> dataSetManager = getDataset(processedFileTable);
+    KeyValueTable keyValueTable = dataSetManager.get();
+    //put record of processed file.
+    File catalogLarge = new File(SOURCE_FOLDER_PATH + "catalogLarge.xml");
```

Comment:
``new File(SOURCE_FOLDER_PATH, "catalogLarge.xml")``. However, you probably will not need it if you choose a different approach per a previous comment.